File: Blind 75/217. Contains Duplicate.ts
// Convert to set
function containsDuplicate(nums: number[]): boolean {
  return nums.length > new Set([...nums]).size;
}

// Using set has
// function containsDuplicate(nums: number[]): boolean {
//   const prevs = new Set();
//   for (const n of nums) {
//     if (prevs.has(n)) return true;
//     prevs.add(n);
//   }
//   return false;
// }

File: Blind Golang/Array/217. Contains Duplicate.go
package array

func containsDuplicate(nums []int) bool {
	prevs := make(map[int]struct{}, len(nums))

	for _, num := range nums {

		if _, ok := prevs[num]; ok {
			return true
		}

		prevs[num] = struct{}{}
	}

	return false
}
